### PR TITLE
Specify SignTransaction's argument types

### LINKED ISSUE #5892: [4.x] signTransactions argument is not typed in web3-types
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

signTransaction in Web3BaseWalletAccount is untyped, or rather accepts anything (https://github.com/web3/web3.js/blob/4.x/packages/web3-types/src/web3_base_wallet.ts#L66). While actual implementation contains acceptable field (https://github.com/web3/web3.js/blob/4.x/packages/web3-eth-accounts/src/account.ts#L244)

### Expected Behavior

Web3BaseWalletAccount should have defined expected parameters

### Steps to Reproduce

...

### Web3.js Version

4.0.5-alpha.5

### Environment

- Operating System: 
- Browser: 
- Node.js Version: 
- NPM Version: 

### Anything Else?

_No response_

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #5892

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

**Review by avkos — CHANGES_REQUESTED** [review]
Please, add information to changelog.md files

**Comment by nikoulai** [comment]
Except for Oleksii's comment LGTM

**Review by nikoulai — APPROVED** [review]

File: packages/web3-eth-contract/CHANGELOG.md [hunk 1]
```diff
@@ -247,6 +247,7 @@ const transactionHash = receipt.transactionHash;
 
 -   `getSendTxParams` will now return `input` instead of `data` in returned transaction parameters object (#5915)
 -   `Contract` constructor will now thrown new `ContractTransactionDataAndInputError` if both `data` and `input` are passed in `ContractInitOptions` for `Contract` constructor (#5915)
+-   The types `ContractAbiWithSignature`, `EventLog`, `ContractEventOptions`, `ContractOptions`, `ContractInitOptions` and `PayableCallOptions` moved to `web3-types`. (#5993)
```
Comment (jdevcs) [inline packages/web3-eth-contract/CHANGELOG.md]: Most of these types are used in contracts package only and not in other packages, except ( `ContractInitOptions` and `PayableCallOptions` ?). so only place shared types in web3-types package. 

Reason: non-shared code will blot other packages as web3-types is included in most of other packages.

**Review by jdevcs — CHANGES_REQUESTED** [review]

**Comment by jdevcs** [comment]
This PR lgtm, except few changes required. Thanks @Muhammad-Altabba

**Comment by jdevcs** [comment]
Some of tests in CI are not passing and there are conflicting files. Could you fix these. Many thanks

**Review by avkos — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Comment by Hasinafidahussain** [comment]
#6050

**Pull request merged**